src/jit/codegenxarch.cpp

```diff
@@ -7549,6 +7556,232 @@ void CodeGen::genSetRegToCond(regNumber dstReg, GenTreePtr tree)
+    // 02D02DF3  jne         02D02DF9
+    // 02D02DF5  pop         ebp
+    // 02D02DF6  ret         8
+    // ; overflow exception
+    // 02D02DF9  call        73BECA1D 
+    emitJumpKind jmpNotEqual = genJumpKindForOper(GT_NE, CK_SIGNED);
+    emitJumpKind jmpLTS = genJumpKindForOper(GT_LT, CK_SIGNED);
+
+    instGen_Compare_Reg_To_Zero(EA_4BYTE, loReg);
+    genJumpToThrowHlpBlk(jmpLTS, SCK_OVERFLOW);
+
+    instGen_Compare_Reg_To_Zero(EA_4BYTE, hiReg);
+    genJumpToThrowHlpBlk(jmpNotEqual, SCK_OVERFLOW);
+}
+
+void CodeGen::genLongToIntCast(GenTreePtr treeNode)
+{
+    assert(treeNode->OperGet() == GT_CAST);
+    assert(treeNode->gtOverflow());
+
+    GenTreePtr  castOp = treeNode->gtCast.CastOp();
+    bool        checkOverflow = treeNode->gtOverflow();
+    var_types   dstType = treeNode->CastToType();
+    bool        isUnsignedDst = varTypeIsUnsigned(dstType);
+    var_types   srcType = genActualType(castOp->TypeGet());
+    bool        isUnsignedSrc = varTypeIsUnsigned(srcType);
+
+    GenTreePtr lowOp = castOp->gtOp.gtOp1;
+    GenTreePtr highOp = castOp->gtOp.gtOp2;
+    regNumber  targetReg = treeNode->gtRegNum;
+    regNumber  sourceReg = lowOp->gtRegNum;
+
+    assert(!varTypeIsSmall(dstType) && "Found directly small type");
+
+    // Narrowing cast
+    noway_assert(genTypeSize(srcType) > genTypeSize(dstType));
+
+    assert(genIsValidIntReg(targetReg));
+    assert(genIsValidIntReg(sourceReg));
+
+    // if necessary, force the srcType to unsigned when the GT_UNSIGNED flag is set
+    if (!isUnsignedSrc && (treeNode->gtFlags & GTF_UNSIGNED) != 0)
+    {
+        srcType = genUnsignedType(srcType);
+        isUnsignedSrc = true;
+    }
+  
+    emitAttr dstSize = EA_ATTR(genTypeSize(dstType));
+    noway_assert(dstSize <= EA_4BYTE);
+
+    genConsumeReg(lowOp);
+    genConsumeReg(highOp);
+
+    if (checkOverflow)
+    {
+        regNumber loReg = sourceReg;
+        regNumber hiReg = highOp->gtRegNum;
+
+        if (isUnsignedSrc)
+        {
+            switch (dstType)
+            {
+            case TYP_UINT:
+                // Same as I8 to U4
+                genOverflowCheck_I8toU4(hiReg, loReg);
+                break;
+            case TYP_INT:
+                genOverflowCheck_U8toI4(hiReg, loReg);
+                break;
+            default:
+                NYI("Cast from long with overflow check");
+            }
+        }
+        else
+        {
+            switch (dstType)
+            {
+            case TYP_INT:
+                genOverflowCheck_I8toI4(hiReg, loReg);
+                break;
+            case TYP_UINT:
+                genOverflowCheck_I8toU4(hiReg, loReg);
+                break;
+            default:
+                NYI("Cast from long with overflow check");
+            }
+        }
+    }
+
+    // Is the value sitting in a non-byte-addressable register? 
+    if (lowOp->InReg() && (dstSize == EA_1BYTE) && !isByteReg(sourceReg))
```

Comment:
This seems to be because RSP, RBP, RSI, and RDI are not byte accessible, so that AND is generated instead of byte-size movzx. I copied this from int to int casting and this looks needed here too. I don't see any conflict with fillPattern. What do you think conflict with EA_1BYTE?